zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1274,7 +1295,6 @@ private void cleanAndNotifyState() {
                 eventThread.queueEvent(new WatchedEvent(Event.EventType.None, Event.KeeperState.Disconnected, null));
```

Comment:
This is the `KeeperState.Disconnected` I mentioned above after disconnected. We should not fire continuous disconnected event without a `KeeperState.SyncConnected`.

Reply:
> We should not fire continuous disconnected event without a `KeeperState.SyncConnected`.

This is not true. Currently, we are firing `KeeperState.Disconnected` after every disconnection even for not established connection.

But it should not affect this pr.